core/activity/src/http_proxy.rs

```diff
@@ -74,8 +80,32 @@ async fn proxy_http_request(
             .call_streaming(msg)
     });
 
+    if let Some(value) = request.headers().get(header::ACCEPT) {
+        if value.eq(mime::TEXT_EVENT_STREAM.essence_str()) {
+            return Ok(Either::Left(stream_results(stream)?));
+        }
+    }
+    Ok(Either::Right(await_results(stream).await?))
+}
+
+fn stream_results(
+    stream: impl Stream<Item = Result<Bytes, Error>> + Unpin + 'static,
+) -> crate::Result<impl Responder> {
     Ok(HttpResponse::Ok()
         .keep_alive()
         .content_type(mime::TEXT_EVENT_STREAM.essence_str())
         .streaming(stream))
 }
+
+async fn await_results(
+    mut stream: impl Stream<Item = Result<Bytes, Error>> + Unpin,
+) -> crate::Result<impl Responder> {
+    let response = stream.next().await;
```

Comment:
Shouldn't we at least check if consecutive `stream.next()` will return `None` and handle `Some` somehow? (error) ?

Reply:
I wasn't sure if it is better to have an error in this case or just assume the first streamed value is OK

Reply:
No idea. Lets leave it as is.